integrationtests/vm_test.go

```diff
@@ -0,0 +1,384 @@
+		assert.NotEmpty(t, publicip)
+
+		node1Container1IP := terraform.Output(t, terraformOptions, "node1_container1_ip")
+		assert.NotEmpty(t, node1Container1IP)
+
+		node1Container2IP := terraform.Output(t, terraformOptions, "node1_container2_ip")
+		assert.NotEmpty(t, node1Container2IP)
+
+		//spliting ip to connect on it
+		pIP := strings.Split(publicip, "/")[0]
+
+		//testing connections
+		ok := TestConnection(pIP, "22")
+		assert.True(t, ok)
+
+	})
+
+	t.Run("single_vm_with_key", func(t *testing.T) {
+		/* Test case for deployeng a singlenode.
+
+		   **Test Scenario**
+
+		   - Deploy a singlenode.
+		   - Check that the outputs not empty.
+		   - connect to the machine
+		   - Destroy the deployment
+
+		*/
+		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: "./singlevm_key_ed25519",
+			Vars: map[string]interface{}{
+				"public_key": publicKey,
+			},
+			Parallelism: 1,
+		})
+		defer terraform.Destroy(t, terraformOptions)
+
+		terraform.InitAndApply(t, terraformOptions)
+
+		// Check that the outputs not empty
+		planetary := terraform.Output(t, terraformOptions, "ygg_ip")
+		assert.NotEmpty(t, planetary)
+
+		node1Container1IP := terraform.Output(t, terraformOptions, "node1_container1_ip")
+		assert.NotEmpty(t, node1Container1IP)
+
+		node1Container2IP := terraform.Output(t, terraformOptions, "node1_container2_ip")
+		assert.NotEmpty(t, node1Container2IP)
+
+		ok := TestConnection(planetary, "22")
+		assert.True(t, ok)
+	})
+
+	t.Run("single_vm_with_zero_CPU", func(t *testing.T) {
+		/* Test case for deployeng a singlenode.
+
+		   **Test Scenario**
+
+		   - Deploy a singlenode with zero cpu.
+		   - The deployment should fail
+		   - Destroy the network
+
+		*/
+		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: "./singlevm_with_zero_cpu",
+			Vars: map[string]interface{}{
+				"public_key": publicKey,
+			},
+			Parallelism: 1,
+		})
+		defer terraform.Destroy(t, terraformOptions)
+
+		_, err = terraform.InitAndApplyE(t, terraformOptions)
+
+		if err == nil {
+			t.Errorf("Should fail with can't deploy with 0 cpu but err is null")
+		}
+	})
+
+	t.Run("single_vm_with_memory_Less_Than_250", func(t *testing.T) {
+		/* Test case for deployeng a singlenode.
+
+		   **Test Scenario**
+
+		   - Deploy a singlenode with memory less than 250.
+		   - The deployment should fail.
+		   - Destroy the network.
+
+		*/
+		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: "./singlevm_with_memory_less_than_250M",
+			Vars: map[string]interface{}{
+				"public_key": publicKey,
+			},
+			Parallelism: 1,
+		})
+		defer terraform.Destroy(t, terraformOptions)
+
+		_, err = terraform.InitAndApplyE(t, terraformOptions)
+
+		if err == nil {
+			t.Errorf("Should fail with mem capacity can't be less that 250M but err is null")
+		}
+
+	})
+	t.Run("qsfs", func(t *testing.T) {
```

Comment:
i think there should be a test here in vm_test.go for mounting a qsfs on a vm, and also a qsfs_test.go file to test qsfs functionality itself.